### PR TITLE
[Listener] Exit process when exception caught in main loop

### PR DESCRIPTION
**Pull request by franckc**
### Description:

Force listener process to restart when catching an exception in its main loop.

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: infra/discovery/src/listener/listener.js [hunk 1]
```diff
@@ -230,9 +230,8 @@ logger.info(
...
-try {
-  main()
-} catch (err) {
-  logger.error('Error occurred in listener main() process')
-  logger.error(err)
-}
+main().catch(err => {
+  logger.error('Error occurred in listener main() process:', err)
+  logger.error('Exiting')
+  process.exit(-1)
```
Comment (mikeshultz) [inline infra/discovery/src/listener/listener.js]: Should probably not use a negative number here.  Can cause unexpected behavior on *nix systems.  Use a number `> 0` to signal an error.

Comment (franckc) [reply infra/discovery/src/listener/listener.js]: good point. done.

**Comment by mikeshultz** [comment]
LGTM, but I don't see an option to approve now?

**Comment by franckc** [comment]
I re-requested your review @mikeshultz - does that work now ?

**Pull request merged**